src/astro/files/base.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pathlib
+from typing import List, Optional
+
+import pandas as pd
+import smart_open
+
+from astro.constants import FileType
+from astro.files.locations import location_factory
+from astro.files.type import type_factory
+
+
+class File:
```

Comment:
Do we really need this class, I thought we just needed FileType and FlieLocation

Reply:
I think it provides nice abstraction, where the only responsibility of client code is to specify path of the file. and everything else is taken care of. 

Also, is there any case where we'll be using location and not types or types and not locations logic separately?

Reply:
@utkarsharma2 @kaxil I agree it would be great to have the `File` abstraction - or something similar to it, e.g.`Dataset`.
When implementing the `Base,Database` class, in parts like: https://github.com/astro-projects/astro/pull/320/files#diff-5280fa37ed21b32d1f80cf6e5768fe10eccf88b0ea0a33b16a3601ca41f82de1R223-R224, I really felt we could benefit by having a central `File` representation which would be used as a container to al file-related bits (location, connection, type, etc). It was not clear to me - for those methods - we benefited by splitting FileType from FileLocation.

Reply:
@utkarsharma2 I had a conversation with @kaxil yesterday about this topic, and he agreed with it :raised_hands: 
I'm marking this conversation as resolved.